src/sentry/services/hybrid_cloud/rpc.py

```diff
@@ -331,6 +335,41 @@ def _create_signatures(cls) -> Mapping[str, RpcMethodSignature]:
                 model_table[base_method.__name__] = signature
         return model_table
 
+    @classmethod
+    def _get_and_validate_local_implementation(cls) -> RpcService:
+        def get_parameters(method: Callable[..., Any]) -> set[str]:
+            """Get the expected set of parameter names.
+
+            The `inspect.signature` also gives us type annotations (on parameters and
+            the return value) but it's tricky to compare those, because of
+            semantically equivalent annotations represented with unequal type tokens,
+            such as `Optional[int]` versus `int | None`.
```

Comment:
We could expand this to check type annotations if we get stricter about exact matches, which is to say, banning the use of `Thing | None` in the `impl` modules and requiring `Optional[Thing]` as we do in the `service` modules. Pro: more checking. Con: moving further away from preferred style.

As I understand it, this problem could go away if we upgrade to Python 3.10+.